### PR TITLE
[wasm] Add support for symbolicating native traces from JS, using a symbols file

### PR DESCRIPTION
**Pull request by radical**
- adds support for symbolicating error, and warning messages
- `debug.js.symbols` is loaded from the vfs
- This is enabled by default, only for non-aot tests (since we don't strip aot tests)
- Works for debugger tests too

### CODE REVIEW

File: src/mono/wasm/runtime/debug.ts [hunk 1]
```diff
@@ -314,25 +334,96 @@ export function mono_wasm_debugger_log(level: number, message_ptr: CharPtr): voi
     console.debug(`Debugger.Debug: ${message}`);
 }
 
+function _readSymbolMapFile(filename: string): void {
+    if (Module.FS_readFile === undefined)
```
Comment (pavelsavara) [inline src/mono/wasm/runtime/debug.ts:338]: That should not really happen, I think.

**Comment by pavelsavara** [comment]
In the future we could also add symbols to stack traces which are marshaled as C# exception.
See https://github.com/dotnet/runtime/blob/842f4d76a8d9344fa6d2e7dbd1534b7b7a6fec57/src/mono/wasm/runtime/method-calls.ts#L459

File: src/mono/wasm/runtime/types/emscripten.ts [hunk 2]
```diff
@@ -49,6 +49,7 @@ export declare interface EmscriptenModule {
     UTF8ArrayToString(u8Array: Uint8Array, idx?: number, maxBytesToRead?: number): string;
     FS_createPath(parent: string, path: string, canRead?: boolean, canWrite?: boolean): string;
     FS_createDataFile(parent: string, name: string, data: TypedArray, canRead: boolean, canWrite: boolean, canOwn?: boolean): string;
+    FS_readFile(filename: string, opts: any): any;
```
Comment (pavelsavara) [inline src/mono/wasm/runtime/types/emscripten.ts:52]: this should force change in `dotnet.d.ts`

**Review by pavelsavara — APPROVED** [review]

**Comment by radical** [comment]
Test failures are unrelated.

**Comment by radical** [comment]
Failures are https://github.com/dotnet/runtime/issues/66231, and https://github.com/dotnet/runtime/issues/65817 .

File: src/mono/sample/wasm/wasm.mk [hunk 3]
```diff
@@ -39,7 +39,7 @@ run-browser:
 	fi
 
 run-console:
-	cd bin/$(CONFIG)/AppBundle && $(V8_PATH) --stack-trace-limit=1000 --single-threaded --expose_wasm $(MAIN_JS) -- $(DOTNET_MONO_LOG_LEVEL) --run $(CONSOLE_DLL) $(ARGS)
```
Comment (radekdoulik) [inline src/mono/sample/wasm/wasm.mk:42]: This broke browser-bench's `make run` for console version. Why was `--run $(CONSOLE_DLL)` removed? Do I need to update browser-bench's Makefile somehow?

**Pull request merged**